pymodule.py

```diff
@@ -38,101 +38,91 @@
+    :param molecule: a Psi4 Molecule object on which computation is performed
+    :param print_warning: Boolean for printing warnings on screen
+    :param kwargs: named arguments associated with Psi4
 
-    # Create the fragment(embedding) projector and apply to rotate orbitals
-    if options.get_bool("EMBEDDING"):
-        forte.print_method_banner(["Frozen-orbital Embedding", "Nan He"])
-        fragment_projector, fragment_nbf = forte.make_fragment_projector(
-            ref_wfn, options)
-        return forte.make_embedding(ref_wfn, options, fragment_projector,
-                                    fragment_nbf, mo_space_info)
+    :return: a Psi4 Wavefunction object from the fresh Psi4 run
+    """
+    ref_type = ref_type.lower().strip()
+
+    if ref_type in ['scf', 'hf', 'rhf', 'rohf', 'uhf']:
+        if print_warning:
+            msg = ['Forte is using orbitals from a Psi4 SCF reference.',
+                   'This is not the best for multireference computations.',
+                   'To use Psi4 CASSCF orbitals, set REF_TYPE to CASSCF.']
+            msg = '\n  '.join(msg)
+            warnings.warn(f"\n  {msg}\n", UserWarning)
+
+        wfn = psi4.driver.scf_helper('forte', molecule=molecule, **kwargs)
+    elif ref_type in ['casscf', 'rasscf']:
+        wfn = psi4.proc.run_detcas(ref_type, molecule=molecule, **kwargs)
     else:
-        return mo_space_info
-
+        raise ValueError(f"Invalid REF_TYPE: {ref_type.upper()} not available!")
 
-def prepare_forte_objects_from_psi4_wfn(options, wfn, mo_space_info):
-    """
-    Take a psi4 wavefunction object and prepare the ForteIntegrals, SCFInfo, and MOSpaceInfo objects
+    return wfn
 
-    Parameters
-    ----------
-    options : ForteOptions
-        A Forte ForteOptions object
-    wfn : psi4 Wavefunction
-        A psi4 Wavefunction object
-    mo_space_info : the MO space info read from options
-        A Forte MOSpaceInfo object
 
-    Returns
-    -------
-    tuple(ForteIntegrals, SCFInfo, MOSpaceInfo)
-        a tuple containing the ForteIntegrals, SCFInfo, and MOSpaceInfo objects
+def check_MO_overlap(options, molecule, Ca):
     """
+    Test the MO overlp matrix is identity or not.
+    We will recompute the SO overlap from molecule and test against Ca,
+    S_MO = Ca^T S_SO Ca.
+    If the molecule and Ca are consistent, S_MO should be identity matrix.
 
-    # Call methods that project the orbitals (AVAS, embedding)
-    mo_space_info = orbital_projection(wfn, options, mo_space_info)
+    :param options: a ForteOptions object
+    :param molecule: a Psi4 Molecule object
+    :param Ca: a Psi4 Matrix that holds orbital coefficients
 
-    # Build Forte SCFInfo object
-    scf_info = forte.SCFInfo(wfn)
-
-    # Build a map from Forte StateInfo to the weights
-    state_weights_map = forte.make_state_weights_map(options, mo_space_info)
-
-    return (state_weights_map, mo_space_info, scf_info)
+    :return: a tuple of (S_MO == I, a new Psi4 Wavefunction object)
```

Comment:
Why are we building a new Psi wavefunction object in our orthogonality checker?

Reply:
You are right, we do not need that. Let me try to directly create a MintsHelper. My previous intent is to always create a new wave function in Forte. But later I feel like it may not be a good idea for finite difference calculations.

Reply:
OK, I think the `std::map<std::string, std::shared_ptr<BasisSet>> basissets_` is not exposed for `MintsHelper`.

Reply:
We should make a PR for this in psi4. I am not sure that X2C modifies the overlap matrix, so maybe we do not need it. But psi4 might automatically try to run X2C (even if not needed) and run into trouble.

Reply:
Yes, maybe in a future PR of fixing the X2C large 1-norm issue.